### PR TITLE
[MM-47081] Migrated "components/searcheable_channel_list_test.jsx" to TypeScript

### PR DESCRIPTION
**Pull request by ridwankabeer435**
#### Summary

Only the corresponding test file `components/searcheable_channel_list_test.jsx` has been converted to TypeScript. `searcheable_channel_list` has already been converted to TypeScript prior to this PR.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/21088
Jira https://mattermost.atlassian.net/browse/MM-47081

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@ridwankabeer435 can you please check failing CI

**Comment by ridwankabeer435** [comment]
So after updating the snapshots, I get to make `searcheable_channel_list_test.tsx` to pass but this is what I get:

> test
> npm run test --workspaces --if-present "webapp/channels/src/components/searchable_channel_list.test.tsx"


> mattermost-webapp@8.0.0 test
> cross-env TZ=Etc/UTC jest "webapp/channels/src/components/searchable_channel_list.test.tsx"

 PASS  src/components/searchable_channel_list.test.tsx
  components/SearchableChannelList
    ✓ should match init snapshot (10 ms)
    ✓ should set page to 0 when starting search (3 ms)

Snapshot Summary
 › 1 snapshot file obsolete from 1 test suite. To remove it, run `npm test -- -u`.
   ↳   • src/components/__snapshots__/searchable_channel_list.test.jsx.snap

Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   1 file obsolete, 1 passed, 1 total
Time:        3.671 s, estimated 4 s
Ran all test suites matching /webapp\/channels\/src\/components\/searchable_channel_list.test.tsx/i.
npm ERR! Lifecycle script `test` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: mattermost-webapp@8.0.0 
npm ERR!   at location: /home/ridk19/go/mattermost-server/webapp/channels 

> @mattermost/client@8.0.0 test
> jest "webapp/channels/src/components/searchable_channel_list.test.tsx"

No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /home/ridk19/go/mattermost-server/webapp/platform/client
  28 files checked.
  testMatch: **/__tests__/**/*.[jt]s?(x), **/?(*.)+(spec|test).[tj]s?(x) - 3 matches
  testPathIgnorePatterns: /node_modules/ - 28 matches
  testRegex:  - 0 matches
Pattern: webapp/channels/src/components/searchable_channel_list.test.tsx - 0 matches
npm ERR! Lifecycle script `test` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @mattermost/client@8.0.0 
npm ERR!   at location: /home/ridk19/go/mattermost-server/webapp/platform/client 

> @mattermost/components@8.0.0 test
> cross-env TZ=Etc/UTC jest "webapp/channels/src/components/searchable_channel_list.test.tsx"

No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /home/ridk19/go/mattermost-server/webapp/platform/components
  179 files checked.
  testMatch: **/__tests__/**/*.[jt]s?(x), **/?(*.)+(spec|test).[tj]s?(x) - 10 matches
  testPathIgnorePatterns: /node_modules/, /.rollup.cache/, /dist/ - 22 matches
  testRegex:  - 0 matches
Pattern: webapp/channels/src/components/searchable_channel_list.test.tsx - 0 matches
npm ERR! Lifecycle script `test` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @mattermost/components@8.0.0 
npm ERR!   at location: /home/ridk19/go/mattermost-server/webapp/platform/components 


In other words, `npm test` is looking for the `searcheable_channel_list_test.tsx` file pattern under `webapp/platform` directory but can't find it. How do we tell npm test to recognize the updated test file?

**Comment by ridwankabeer435** [comment]
However, the test for the component passes upon running '''make test'''.

**Comment by M-ZubairAhmed** [comment]
> However, the test for the component passes upon running `make test`.

Yup this works too

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Check you CI, it mentions how to remove the obsolete test snapshot

**Comment by ridwankabeer435** [comment]
> 

Done

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
@ridwankabeer435  thank you so much again for contributing to mattermost with this PR. Your contribution is very valuable. 💙 🎊 We appreciate your help to help make better software.
